File: src/main/java/ru/citeck/idea/references/resolvers/YAMLScalarValueResolver.java
package ru.citeck.idea.references.resolvers;

import org.jetbrains.yaml.psi.YAMLScalar;
import ru.citeck.idea.references.PsiElementValueResolver;

public class YAMLScalarValueResolver implements PsiElementValueResolver<YAMLScalar> {

    @Override
    public Class<YAMLScalar> getPsiElementType() {
        return YAMLScalar.class;
    }

    @Override
    public Object getValue(YAMLScalar psiElement) {
        return psiElement.getText();
    }

}

File: src/main/kotlin/ru/citeck/idea/references/resolvers/YAMLScalarValueResolver.kt
package ru.citeck.idea.references.resolvers

import org.jetbrains.yaml.psi.YAMLScalar
import ru.citeck.idea.references.PsiElementValueResolver

class YAMLScalarValueResolver : PsiElementValueResolver<YAMLScalar> {

    override fun getPsiElementType(): Class<YAMLScalar> {
        return YAMLScalar::class.java
    }

    override fun getValue(psiElement: YAMLScalar): Any? {
        return psiElement.text
    }
}
